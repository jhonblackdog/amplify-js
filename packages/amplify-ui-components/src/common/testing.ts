export const pixelThreshold = 10;
