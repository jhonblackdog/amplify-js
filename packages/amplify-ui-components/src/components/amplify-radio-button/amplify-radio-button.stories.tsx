import { storiesOf } from '@storybook/html';
import { text as textKnob, boolean as booleanKnob } from '@storybook/addon-knobs';

const radioButtonStories = storiesOf('amplify-radio-button', module);

radioButtonStories.add('with label', () => {
  const label = textKnob('Label', 'Seattle');

  return `<amplify-radio-button label=${label} field-id="seattle" name="seattle"></amplify-radio-button>`;
});

radioButtonStories.add('checkable radio button', () => {
  const label = textKnob('Label', 'Oceanside');
  const toggle = booleanKnob('Toggle', false);

  return `<amplify-radio-button label=${label} field-id="oceanside" name="oceanside" checked=${toggle}></amplify-radio-button>`;
});

radioButtonStories.add('disabled', () => {
  const label = textKnob('Label', 'Moscow');
  const toggle = booleanKnob('Disabled', true);

  return `<amplify-radio-button label=${label} field-id="moscow" name="moscow" disabled=${toggle}></amplify-radio-button>`;
});
