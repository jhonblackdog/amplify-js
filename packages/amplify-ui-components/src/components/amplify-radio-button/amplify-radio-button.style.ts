import { css } from 'emotion';

export const radioButton = css`
  margin-bottom: 22px;
  display: block;
  width: 100%;
  padding: 16px;
  font-size: 14px;
  font-family: var(--font-family);
`;