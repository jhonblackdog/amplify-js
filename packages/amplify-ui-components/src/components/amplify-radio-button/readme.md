# amplify-radio-button



<!-- Auto Generated Below -->


## Properties

| Property        | Attribute        | Description                                           | Type      | Default     |
| --------------- | ---------------- | ----------------------------------------------------- | --------- | ----------- |
| `fieldId`       | `field-id`       | Field ID used for the 'for' in the label              | `string`  | `undefined` |
| `label`         | `label`          | Label for the radio button                            | `string`  | `undefined` |
| `name`          | `name`           | (optional) Name of radio button                       | `string`  | `undefined` |
| `styleOverride` | `style-override` | (optional) Whether or not to override default styling | `boolean` | `false`     |
| `type`          | `type`           | Type of input for this component is radio             | `string`  | `'radio'`   |
| `value`         | `value`          | (optional) Value of radio button                      | `string`  | `undefined` |


## Dependencies

### Used by

 - [amplify-examples](../amplify-examples)

### Depends on

- [amplify-label](../amplify-label)

### Graph
```mermaid
graph TD;
  amplify-radio-button --> amplify-label
  amplify-examples --> amplify-radio-button
  style amplify-radio-button fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
