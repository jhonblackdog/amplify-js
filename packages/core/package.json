{
  "name": "@aws-amplify/core",
  "version": "1.0.26-unstable.1",
  "description": "Core category of aws-amplify",
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/src/index.js",
  "typings": "./dist/esm/src/index.d.ts",
  "sideEffects": false,
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "test": "tslint 'src/**/*.ts' && jest --coverage",
    "build-with-test": "npm test && npm run build",
    "build:cjs": "tsc -p tsconfig.es5.json && rollup -c",
    "build:esm": "tsc -p tsconfig.es2015.json",
    "build": "npm run clean && npm run build:cjs && npm run build:esm",
    "clean": "rimraf tsc-out dist",
    "format": "tsfmt --useTsfmt tsfmt.json -r src/**/*.ts",
    "lint": "tslint 'src/**/*.ts'"
  },
  "react-native": {
    "./lib/ClientDevice": "./lib/ClientDevice/reactnative.js",
    "./lib/RNComponents": "./lib/RNComponents/reactnative.js",
    "./lib/StorageHelper": "./lib/StorageHelper/reactnative.js"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/aws/aws-amplify.git"
  },
  "author": "Amazon Web Services",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/aws/aws-amplify/issues"
  },
  "homepage": "https://github.com/aws/aws-amplify#readme",
  "devDependencies": {
    "find": "^0.2.7",
    "prepend-file": "^1.3.1",
    "prettier": "^1.7.4",
    "typescript": "^3.2.2"
  },
  "dependencies": {
    "aws-sdk": "2.329.0",
    "url": "^0.11.0"
  },
  "jest": {
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": "<rootDir>../../node_modules/ts-jest/preprocessor.js"
    },
    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(tsx?|jsx?)$",
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "json",
      "jsx"
    ],
    "testEnvironment": "jsdom",
    "mapCoverage": true,
    "coverageThreshold": {
      "global": {
        "branches": 0,
        "functions": 0,
        "lines": 0,
        "statements": 0
      }
    },
    "testURL": "http://localhost/",
    "coveragePathIgnorePatterns": [
      "/node_modules/"
    ]
  }
}
