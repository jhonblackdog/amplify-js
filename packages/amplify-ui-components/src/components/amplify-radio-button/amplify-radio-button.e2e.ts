import { newE2EPage } from '@stencil/core/testing';

describe('Radio Button Component:', () => {
  it('should render a amplify-radio-button', async () => {
    const page = await newE2EPage();
    await page.setContent(`<amplify-radio-button></amplify-radio-button>`);
    const el = await page.find('amplify-radio-button');
    expect(el).not.toBeNull();
  });
});