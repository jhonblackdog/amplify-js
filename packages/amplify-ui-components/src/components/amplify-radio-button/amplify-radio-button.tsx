import { Component, Prop, h } from '@stencil/core';
import { radioButton } from './amplify-radio-button.style';
import { styleNuker } from '../../common/helpers';
import { AMPLIFY_UI_PREFIX } from '../../common/constants';

const STATIC_RADIO_BUTTON_CLASS_NAME = `${AMPLIFY_UI_PREFIX}--radio-button`;

@Component({
  tag: 'amplify-radio-button',
  shadow: false,
})
export class AmplifyRadioButton {
  /** (Optional) Overrides default styling */
  @Prop() styleOverride?: boolean = false;
  /** (Optional) Name of radio button */
  @Prop() name?: string;
  /** (Optional) Value of radio button */
  @Prop() value?: string;
  /** Field ID used for the 'for' in the label */
  @Prop() fieldId: string;
  /** Label for the radio button */
  @Prop() label: string;
  /** If `true`, the radio button is selected. */
  @Prop() checked: boolean = false;
  /** If `true`, the checkbox is disabled */
  @Prop() disabled: boolean = false;

  render() {
    return (
      <span class={styleNuker(this.styleOverride, STATIC_RADIO_BUTTON_CLASS_NAME, radioButton)}>
        <input
          type="radio"
          name={this.name}
          value={this.value}
          id={this.fieldId}
          checked={this.checked}
          disabled={this.disabled}
        />
        <amplify-label htmlFor={this.fieldId}>{this.label}</amplify-label>
      </span>
    );
  }
}