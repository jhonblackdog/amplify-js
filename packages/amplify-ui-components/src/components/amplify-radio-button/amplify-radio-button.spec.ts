import { newSpecPage } from '@stencil/core/testing';
import { AmplifyRadioButton } from './amplify-radio-button';

describe('amplify-radio-button:', () => {
  describe('Default cases ->', () => {
    describe('Class logic', () => {
      let radio;

      beforeEach(() => {
        radio = new AmplifyRadioButton();
      });

      it('should render false for checked prop', () => {
        expect(radio.checked).toBe(false);
      });

      it('should render false for disabled prop', () => {
        expect(radio.disabled).toBe(false);
      });

      it('should render false for style override prop', () => {
        expect(radio.styleOverride).toBe(false);
      });
    });
    describe('HTML logic', () => {
      it('should render only class `amplify-ui--radio-button` when style override is set to true', async () => {
        const page = await newSpecPage({
          components: [AmplifyRadioButton],
          html: `<amplify-radio-button style-override="true"></amplify-radio-button>`
        });

        expect(page.root).toMatchSnapshot();
      });
      it('should render a radio button with no label and a type of radio for input', async () => {
        const page = await newSpecPage({
          components: [AmplifyRadioButton],
          html: `<amplify-radio-button></amplify-radio-button>`
        });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});