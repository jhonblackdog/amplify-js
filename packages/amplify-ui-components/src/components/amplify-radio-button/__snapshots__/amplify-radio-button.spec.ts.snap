// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-radio-button: Default cases -> HTML logic should render a radio button with no label and a type of radio for input 1`] = `
<amplify-radio-button>
  <span class="amplify-ui--radio-button css-iud8ha">
    <input type="radio">
    <amplify-label></amplify-label>
  </span>
</amplify-radio-button>
`;

exports[`amplify-radio-button: Default cases -> HTML logic should render only class \`amplify-ui--radio-button\` when style override is set to true 1`] = `
<amplify-radio-button style-override="true">
  <span class="amplify-ui--radio-button">
    <input type="radio">
    <amplify-label></amplify-label>
  </span>
</amplify-radio-button>
`;
